src/document.rs

```diff
@@ -276,3 +304,857 @@ impl Display for Created {
+
+    /// This field provides a place for the SPDX file creator to record file contributors.
+    /// Contributors could include names of copyright holders and/or authors who may not be
+    /// copyright holders yet contributed to the file content.
+    #[serde(rename = "fileContributors", skip_serializing_if = "Option::is_none")]
+    pub file_contributors: Option<Vec<String>>,
+
+    #[serde(rename = "fileDependencies", skip_serializing_if = "Option::is_none")]
+    pub file_dependencies: Option<Vec<String>>,
+
+    /// The name of the file relative to the root of the package.
+    #[serde(rename = "fileName")]
+    pub file_name: String,
+
+    /// The type of the file.
+    #[serde(rename = "fileTypes", skip_serializing_if = "Option::is_none")]
+    pub file_types: Option<Vec<FileType>>,
+
+    /// The licenseComments property allows the preparer of the SPDX document to describe why the
+    /// licensing in spdx:licenseConcluded was chosen.
+    #[serde(rename = "licenseComments", skip_serializing_if = "Option::is_none")]
+    pub license_comments: Option<String>,
+
+    /// License expression for licenseConcluded.  The licensing that the preparer of this SPDX
+    /// document has concluded, based on the evidence, actually applies to the package.
+    #[serde(rename = "licenseConcluded")]
+    pub license_concluded: String,
+
+    /// Licensing information that was discovered directly in the subject file. This is also
+    /// considered a declared license for the file.
+    #[serde(rename = "licenseInfoInFiles", skip_serializing_if = "Option::is_none")]
+    pub license_info_in_files: Option<Vec<String>>,
+
+    /// This field provides a place for the SPDX file creator to record potential legal notices
+    /// found in the file. This may or may not include copyright statements.
+    #[serde(rename = "noticeText", skip_serializing_if = "Option::is_none")]
+    pub notice_text: Option<String>,
+
+    /// Uniquely identify any element in an SPDX document which may be referenced by other
+    /// elements.
+    #[serde(rename = "SPDXID")]
+    pub spdxid: String,
+}
+
+impl File {
+    pub fn try_from_binary(path: &Utf8PathBuf) -> Result<File> {
+        let file_name = path.file_name().unwrap();
+        let spdxid = format!("SPDXRef-File-{}", file_name);
+        Ok(File {
+            annotations: None,
+            attribution_texts: None,
+            checksums: Some(calculate_checksums(path)?),
+            comment: None,
+            copyright_text: NOASSERTION.to_string(),
+            file_contributors: None,
+            file_dependencies: None,
+            file_name: file_name.to_string(),
+            file_types: Some(vec![FileType::Binary]),
+            license_comments: None,
+            license_concluded: NOASSERTION.to_string(),
+            license_info_in_files: None,
+            notice_text: None,
+            spdxid,
+        })
+    }
+}
+
+/// Generate SHA1 and SHA256 checksums for a given file
```

Comment:
Thought it was reasonable to generate SHA256 in addition to SHA1 for users worried about SHA1 usage.

Reply:
Yeah absolutely. In the GitBOM context, and with other SBOM tools I've seen, doing both has been the norm.